src/include/stir/recon_buildblock/DataSymmetriesForBins_PET_CartesianGrid.inl

```diff
@@ -89,11 +97,56 @@ find_transform_z(
 				     + num_planes_per_scanner_ring*delta
 				     + 2*axial_pos_to_z_offset[segment_num]); 
     // now use rounding to be safe
-    int transform_z = (int)floor(transform_z_float + 0.5);
+    transform_z = (int)floor(transform_z_float + 0.5);
     assert(fabs(transform_z-transform_z_float) < 10E-4);
 
-    return transform_z;
   }
+      }
+    //block implementaion
+    if (proj_data_info_ptr->get_scanner_ptr()->get_scanner_geometry()=="BlocksOnCylindrical")
+      {
+  const ProjDataInfoBlocksOnCylindrical* proj_data_info_blk_ptr =
+      static_cast<const ProjDataInfoBlocksOnCylindrical *>(proj_data_info_ptr.get());
+
+    const float delta = proj_data_info_blk_ptr->get_average_ring_difference(segment_num);
+
+
+    // Find symmetric value in Z by 'mirroring' it around the centre z of the LOR:
+    // Z+Q = 2*centre_of_LOR_in_image_coordinates == transform_z
+    {
+      // first compute it as floating point (although it has to be an int really)
+      const float transform_z_float = (2*num_planes_per_axial_pos[segment_num]*(axial_pos_num)
+               + num_planes_per_scanner_ring*delta
+               + 2*axial_pos_to_z_offset[segment_num]);
+      // now use rounding to be safe
+      transform_z = (int)floor(transform_z_float + 0.5);
+      assert(fabs(transform_z-transform_z_float) < 10E-4);
+
+    }
+      }
+    // generic implementation
+    else
+    {
+        const ProjDataInfoGeneric* proj_data_info_gen_ptr =
+        static_cast<const ProjDataInfoGeneric *>(proj_data_info_ptr.get());
+
+        const float delta = proj_data_info_gen_ptr->get_average_ring_difference(segment_num);
```

Comment:
When running in debug mode  the test_DataSymmetriesForBins_PET_CartesianGrid we get the vectorWithOffset assertion "begin_allocated_memory <= num+start" it seems the min_ring_diff and max_ring_diff indeces go from 0 to 4 while segment_num is 5